src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java

```diff
@@ -48,19 +48,19 @@ enum Kind {
     }
 
     private final SourceFileBuilder sb;
-    private final List<String> enclosingClassNames;
     private final String modifiers;
     private final Kind kind;
     private final String className;
     private final String superName;
+    private final ClassSourceBuilder enclosing;
 
-    ClassSourceBuilder(SourceFileBuilder builder, String modifiers, Kind kind, String className, String superName, List<String> enclosingClassNames) {
+    ClassSourceBuilder(SourceFileBuilder builder, String modifiers, Kind kind, String className, String superName, ClassSourceBuilder enclosing) {
         this.sb = builder;
         this.modifiers = modifiers;
         this.kind = kind;
         this.className = className;
         this.superName = superName;
-        this.enclosingClassNames = enclosingClassNames;
+        this.enclosing = enclosing;
```

Comment:
should we check that the enclosing class has the same source file builder as this one?

Reply:
Note really important from the `ClassSourceBuilder` perspective. Having builders with different underlying `SourceFileBuilders` would still work, the sources would just be emitted to different source files.